### PR TITLE
CUMULUS-3104-2: Fixed TS compilation error caused by @aws-sdk/client-s3 3.202.0 upgrade

### PR DESCRIPTION
**Pull request by jennyhliu**
**Summary:** Summary of changes

Addresses [CUMULUS-XX: Develop amazing new feature](https://bugs.earthdata.nasa.gov/browse/CUMULUS-XXX)

## Changes

```
lerna ERR! npm run tsc:listEmittedFiles exited 2 in '@cumulus/aws-client'

src/SQS.ts:163:14 - error TS2345: Argument of type 'string | undefined' is not assignable to parameter of type 'string'.
  Type 'undefined' is not assignable to type 'string'.
163   JSON.parse(get(message, 'Body', get(message, 'body', '{}'))); 
```

### CODE REVIEW

File: packages/aws-client/src/SQS.ts [hunk 1]
```diff
@@ -160,7 +160,7 @@ export const receiveSQSMessages = async (
 export const parseSQSMessageBody = (
   message: SQSRecord | AWS.SQS.Message
 ): { [key: string]: any } =>
-  JSON.parse(get(message, 'Body', get(message, 'body', '{}')));
+  JSON.parse(get(message, 'Body', get(message, 'body')) || '{}');
```
Comment (Jkovarik) [inline packages/aws-client/src/SQS.ts:163]: Did lodash.get change?        Why was this needed?

**Comment by Jkovarik** [comment]
For posterity, this is to resolve:

```
src/SQS.ts:163:14 - error TS2345: Argument of type 'string | undefined' is not assignable to parameter of type 'string'.
  Type 'undefined' is not assignable to type 'string'.
163   JSON.parse(get(message, 'Body', get(message, 'body', '{}'))); 
```

File: example/spec/parallel/discoverGranules/DiscoverGranulesDuplicateHandlingErrorSpec.js [hunk 2]
```diff
@@ -164,7 +164,7 @@ describe('The DiscoverGranules workflow with an existing granule and duplicateHa
         status: 'failed',
       });
 
-      const errorCause = JSON.parse(get(execution, 'error.Cause', {}));
+      const errorCause = JSON.parse(get(execution, 'error.Cause', '{}'));
```
Comment (Jkovarik) [inline example/spec/parallel/discoverGranules/DiscoverGranulesDuplicateHandlingErrorSpec.js:167]: Good eye!

File: packages/aws-client/src/SQS.ts (hunk 1 above)
Comment (Jkovarik) [inline packages/aws-client/src/SQS.ts:163]: Nit: we should use nullish coalescing here instead of || generally as a pattern. 

Something like JSON.parse(message?.Body?.body ?? '{}')

**Review by Jkovarik — CHANGES_REQUESTED** [review]
Looks good, just a comment about || vs ??.

File: packages/aws-client/src/SQS.ts (hunk 1 above)
Comment (Jkovarik) [reply packages/aws-client/src/SQS.ts:163]: Apologies, I misread that on the first pass.   

Something like `get(message, 'Body', get(message, 'body')) ?? '{}'`?    The important bit is `||` vs `??`.

**Review by Jkovarik — APPROVED** [review]
Looks good!

File: example/spec/parallel/ingestGranule/IngestGranuleFailureSpec.js [hunk 3]
```diff
@@ -107,6 +107,7 @@ describe('The Ingest Granule failure workflow', () => {
         },
       });
     } catch (error) {
+      console.log(`*** error.apiMessage ${error.apiMessage}`);
```
Comment (jennyhliu) [inline example/spec/parallel/ingestGranule/IngestGranuleFailureSpec.js:110]: Add debug for ci failure:
```
An error was thrown in an afterAll
SyntaxError: Unexpected token u in JSON at position 0
at JSON.parse (<anonymous>)
at UserContext.<anonymous> /cumulus/example/spec/parallel/ingestGranule/IngestGranuleFailureSpec.js:110:31
```
